be/src/exprs/vectorized/like_predicate.cpp

```diff
@@ -406,42 +421,57 @@ ColumnPtr LikePredicate::regex_match_full(FunctionContext* context, const starro
 
 ColumnPtr LikePredicate::regex_match_partial(FunctionContext* context, const starrocks::vectorized::Columns& columns) {
     auto value_column = VECTORIZED_FN_ARGS(0);
+    auto pattern_column = VECTORIZED_FN_ARGS(1);
 
     ColumnViewer<TYPE_VARCHAR> value_viewer(value_column);
     ColumnBuilder<TYPE_BOOLEAN> result(value_viewer.size());
 
     // pattern is constant value, use context's regex
     if (context->is_constant_column(1)) {
-        auto state = reinterpret_cast<LikePredicateState*>(context->get_function_state(FunctionContext::THREAD_LOCAL));
-
-        hs_scratch_t* scratch = nullptr;
-        if (hs_clone_scratch(state->scratch, &scratch) != HS_SUCCESS) {
-            CHECK(false) << "ERROR: Unable to clone scratch space.";
-        }
+        if (!pattern_column->is_nullable()) {
+            auto state =
+                    reinterpret_cast<LikePredicateState*>(context->get_function_state(FunctionContext::THREAD_LOCAL));
+
+            hs_scratch_t* scratch = nullptr;
+            hs_error_t status;
+            if ((status = hs_clone_scratch(state->scratch, &scratch)) != HS_SUCCESS) {
+                CHECK(false) << "ERROR: Unable to clone scratch space."
+                             << " status: " << status;
+            }
 
-        for (int row = 0; row < value_viewer.size(); ++row) {
-            bool v = false;
-            auto status = hs_scan(
-                    state->database, value_viewer.value(row).data, value_viewer.value(row).size, 0, scratch,
-                    [](unsigned int id, unsigned long long from, unsigned long long to, unsigned int flags,
-                       void* ctx) -> int {
-                        *((bool*)ctx) = true;
-                        return 1;
-                    },
-                    &v);
-
-            DCHECK(status == HS_SUCCESS || status == HS_SCAN_TERMINATED);
-            result.append(v, value_viewer.is_null(row));
-        }
+            for (int row = 0; row < value_viewer.size(); ++row) {
+                if (value_viewer.is_null(row)) {
+                    result.append_null();
+                    continue;
+                }
+
+                bool v = false;
+                auto value_size = value_viewer.value(row).size;
+                auto status = hs_scan(
+                        state->database, (value_size) ? value_viewer.value(row).data : &_DUMMY_STRING_USE_WITH_HS_SCAN,
+                        value_size, 0, scratch,
+                        [](unsigned int id, unsigned long long from, unsigned long long to, unsigned int flags,
+                           void* ctx) -> int {
+                            *((bool*)ctx) = true;
+                            return 1;
+                        },
+                        &v);
+
+                DCHECK(status == HS_SUCCESS || status == HS_SCAN_TERMINATED) << " status: " << status;
+                result.append(v);
```

Comment:
line 431~468,  a new function name _predicate_const_regex function may be introduced to simplify the code and make the code clear and understandable.